easypost/shipment.py

```diff
@@ -1,65 +1,59 @@
-        self.refresh_from(response, api_key)
+        response, api_key = requestor.request(method="post", url=url)
+        self.refresh_from(values=response, api_key=api_key)
         return self
 
-    def get_smartrates(self):
-        requestor = Requestor(self._api_key)
+    def get_smartrates(self) -> List[object]:
+        requestor = Requestor(local_api_key=self._api_key)
         url = "%s/%s" % (self.instance_url(), "smartrate")
-        response, api_key = requestor.request("get", url)
+        response, api_key = requestor.request(method="get", url=url)
         return response.get("result", [])
 
-    def buy(self, **params):
-        requestor = Requestor(self._api_key)
+    def buy(self, **params) -> "Shipment":
+        requestor = Requestor(local_api_key=self._api_key)
         url = "%s/%s" % (self.instance_url(), "buy")
-        response, api_key = requestor.request("post", url, params)
-        self.refresh_from(response, api_key)
+        response, api_key = requestor.request(method="post", url=url, params=params)
+        self.refresh_from(values=response, api_key=api_key)
         return self
 
-    def refund(self, **params):
-        requestor = Requestor(self._api_key)
+    def refund(self, **params) -> "Shipment":
+        requestor = Requestor(local_api_key=self._api_key)
         url = "%s/%s" % (self.instance_url(), "refund")
-        response, api_key = requestor.request("post", url, params)
-        self.refresh_from(response, api_key)
+        response, api_key = requestor.request(method="post", url=url, params=params)
+        self.refresh_from(values=response, api_key=api_key)
         return self
 
-    def insure(self, **params):
-        requestor = Requestor(self._api_key)
+    def insure(self, **params) -> "Shipment":
+        requestor = Requestor(local_api_key=self._api_key)
         url = "%s/%s" % (self.instance_url(), "insure")
-        response, api_key = requestor.request("post", url, params)
-        self.refresh_from(response, api_key)
+        response, api_key = requestor.request(method="post", url=url, params=params)
+        self.refresh_from(values=response, api_key=api_key)
         return self
 
-    def label(self, **params):
-        requestor = Requestor(self._api_key)
+    def label(self, **params) -> "Shipment":
+        requestor = Requestor(local_api_key=self._api_key)
         url = "%s/%s" % (self.instance_url(), "label")
-        response, api_key = requestor.request("get", url, params)
-        self.refresh_from(response, api_key)
+        response, api_key = requestor.request(method="get", url=url, params=params)
+        self.refresh_from(values=response, api_key=api_key)
         return self
 
-    def lowest_rate(self, carriers=None, services=None):
+    def lowest_rate(self, carriers: List[str] = None, services: List[str] = None) -> Rate:
         carriers = carriers or []
         services = services or []
         lowest_rate = None
 
-        try:
```

Comment:
Hmm... agree that the removal of this is a good thing because it's supposed to be a list, but it's still a breaking change that should get called out in the CHANGELOG in case someone was relying on this.

Reply:
Resolved in 94d6bd1b26f8d68dcb940fd8f3f37d758f270568